go/indexer/confirmed_headers.go

```diff
@@ -0,0 +1,137 @@
+package indexer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+const (
+	DefaultConnectionTimeout        = 20 * time.Second
+	DefaultConfDepth         uint64 = 20
+	DefaultMaxBatchSize      uint64 = 100
+)
+
+type HeaderSelector interface {
+	NewHead(context.Context, uint64, *types.Header, HeaderBackend) []*types.Header
+}
+
+type HeaderBackend interface {
+	HeaderByNumber(context.Context, *big.Int) (*types.Header, error)
+}
+
+type HeaderSelectorConfig struct {
+	ConfDepth    uint64
+	MaxBatchSize uint64
+}
+
+type ConfirmedHeaderSelector struct {
+	cfg HeaderSelectorConfig
+}
+
+func (f *ConfirmedHeaderSelector) NewHead(
+	ctx context.Context,
+	lowest uint64,
+	header *types.Header,
+	backend HeaderBackend,
+) []*types.Header {
+
+	number := header.Number.Uint64()
+	blockHash := header.Hash()
+
+	fmt.Printf("🎁 New block=%d hash=%s\n", number, blockHash)
```

Comment:
The `fmt.Printf` calls should use the logger.

Reply:
Fixed.